05.wc/wc.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options, filenames = parse_options
+  file_stats = collect_file_stats(filenames)
+  total_stat = calculate_total_stat(file_stats)
+  max_widths = calculate_max_widths(total_stat)
+
+  file_stats << total_stat if filenames.size > 1
+  file_stats.each { |file_stat| puts format_result(file_stat, max_widths, options).join(' ') }
```

Comment:
```suggestion
  file_stats.each { |file_stat| puts format_row(file_stat, max_widths, options) }
```

joinはメソッドの中でやってしまって、「行をフォーマットするメソッド」とするのはどうでしょう？
配列を戻り値にしても、誰も喜ばない（配列として扱いたい処理がない）ので。